fvm/evm/stdlib/contract_test.go

```diff
@@ -809,3 +815,101 @@ func TestEVMAddressDeposit(t *testing.T) {
 
 	require.True(t, deposited)
 }
+
+func TestBridgedAccountWithdraw(t *testing.T) {
+
+	t.Parallel()
+
+	var withdrew bool
+
+	contractsAddress := flow.BytesToAddress([]byte{0x1})
+
+	handler := &testContractHandler{
+		flowTokenAddress: common.Address(contractsAddress),
+		accountByAddress: func(fromAddress types.Address, isAuthorized bool) types.Account {
+			assert.Equal(t, types.Address{1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, fromAddress)
+			assert.True(t, isAuthorized)
+
+			return &testFlowAccount{
+				address: fromAddress,
+				withdraw: func(balance types.Balance) *types.FLOWTokenVault {
+					assert.Equal(t, types.Balance(0), balance)
+					withdrew = true
+					return types.NewFlowTokenVault(balance)
+				},
+			}
+		},
+	}
+
+	env := runtime.NewBaseInterpreterEnvironment(runtime.Config{})
+
+	stdlib.SetupEnvironment(env, handler, contractsAddress)
+
+	rt := runtime.NewInterpreterRuntime(runtime.Config{})
+
+	script := []byte(`
+      import EVM from 0x1
+      import FlowToken from 0x1
+
+      access(all)
+      fun main(): UFix64 {
+          let bridgedAccount <- EVM.createBridgedAccount()
+          let vault <- bridgedAccount.withdraw(balance: EVM.Balance(flow: 0.0))
```

Comment:
does destroying a flow vault impact the total token count stored on the Flow token smart contract?

Reply:
Yes! This is why the tests deposit/withdraw 0 for now. We still need to properly mint tokens in the test. When I didn't mint tokens and tried to depositing a non-zero amount, the total supply was underflowing :D